cpp/convert_json_to_marc.cc

```diff
@@ -0,0 +1,493 @@
+
+    const auto array_node(JSON::JSONNode::CastToArrayNodeOrDie("string_or_array_node", string_or_array_node));
+    for (const auto element_node : *array_node) {
+        const auto bibliographic_level(getBibliographicLevel(JSON::JSONNode::CastToStringNodeOrDie("element_node", element_node)->getValue()));
+        if (bibliographic_level != default_)
+            return bibliographic_level;
+    }
+
+    return default_;
+}
+
+
+void ProcessGlobalSection(const IniFile::Section &global_section, std::string * const root_path,
+                          JSONNodeToBibliographicLevelMapper ** const json_node_to_bibliographic_level_mapper)
+{
+    *root_path = global_section.getString("root_path");
+
+    if (global_section.hasEntry("item_type_tag") and not global_section.hasEntry("item_type_map"))
+        LOG_ERROR("Global section \"has item_type_tag\" but not \"item_type_map\"!");
+
+    if (not global_section.hasEntry("item_type_tag") and global_section.hasEntry("item_type_map"))
+        LOG_ERROR("Global section \"has item_type_map\" but not \"item_type_tag\"!");
+
+    std::string item_type_tag, item_type_map;
+    if (global_section.hasEntry("item_type_tag")) {
+        item_type_tag = global_section.getString("item_type_tag");
+        item_type_map = global_section.getString("item_type_map");
+    }
+
+    *json_node_to_bibliographic_level_mapper = new JSONNodeToBibliographicLevelMapper(item_type_tag, item_type_map);
+}
+
+
+std::vector<FieldDescriptor> LoadFieldDescriptors(const std::string &inifile_path, std::string * const root_path,
+                                                  JSONNodeToBibliographicLevelMapper ** const json_node_to_bibliographic_level_mapper)
+{
+    std::vector<FieldDescriptor> field_descriptors;
+
+    const IniFile ini_file(inifile_path);
+    for (const auto &section : ini_file) {
+        const auto &section_name(section.getSectionName());
+        if (section_name.empty())
+            continue;
+
+        if (section_name == "Global")
+            ProcessGlobalSection(section, root_path, json_node_to_bibliographic_level_mapper);
+        else { // A section describing a mapping to a field
+            const auto tag(section.getString("tag", ""));
+            if (tag.empty())
+                LOG_ERROR("missing tag in section \"" + section_name + "\" in \"" + ini_file.getFilename() + "\"!");
+            if (tag.length() != MARC::Record::TAG_LENGTH)
+                LOG_ERROR("invalid tag \"" + tag + "\" in section \"" + section_name + "\" in \"" + ini_file.getFilename() + "\"!");
+
+            std::vector<std::pair<char, std::string>> subfield_codes_to_json_tags, subfield_codes_to_prefixes, subfield_codes_to_fixed_subfields;
+            for (const auto section_entry : section) {
+                if (StringUtil::StartsWith(section_entry.name_, "add_fixed_subfield_")) {
+                    if (section_entry.name_.length() != __builtin_strlen("add_fixed_subfield_?"))
```

Comment:
Perhaps add a comment here to explain that the '?' character is a placeholder, or just add 1 to the prefix's length.

Reply:
Done.